File: app/src/main/java/com/android/granolasstore/ShoppingCart.java
package com.android.granolasstore;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import java.util.ArrayList;

public class ShoppingCart extends AppCompatActivity {

    private RecyclerView mShoppingCartRecyclerView;
    private ShoppingCartAdapter mShoppingCartAdapter;

    private TextView mNoItemsText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_shopping_cart);

        mShoppingCartRecyclerView = findViewById(R.id.shopping_cart_recyclerview);
        mNoItemsText = findViewById(R.id.no_items_text);

        Intent intent = getIntent();
        ArrayList<String> shoppingCart
                = (ArrayList<String>)intent.getSerializableExtra(MainActivity.EXTRA_SHOPPING_CART);

        if (shoppingCart.size() > 0) {
            mNoItemsText.setVisibility(View.GONE);
        } else {
            mNoItemsText.setVisibility(View.VISIBLE);
        }

        mShoppingCartAdapter = new ShoppingCartAdapter(this, shoppingCart);
        mShoppingCartRecyclerView.setAdapter(mShoppingCartAdapter);
        mShoppingCartRecyclerView.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: app/src/main/java/com/android/granolasstore/ShoppingCart.kt
package com.android.granolasstore

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.LinearLayoutManager
import com.android.granolasstore.databinding.FragmentShoppingCartBinding
import java.util.ArrayList

class ShoppingCart : Fragment() {

    private var _binding: FragmentShoppingCartBinding? = null
    private val binding get() = _binding!!

    var a: Array<String>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        arguments?.let {
            a = it.getStringArray("shoppingCart")!!
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        _binding = FragmentShoppingCartBinding.inflate(inflater, container, false)

        if (a != null) {
            val shoppingCartAdapter = ShoppingCartAdapter(context, a!!)
            binding.shoppingCartRecyclerview.adapter = shoppingCartAdapter
            binding.shoppingCartRecyclerview.layoutManager = LinearLayoutManager(context)
        } else {
            Toast.makeText(context, "f", Toast.LENGTH_SHORT).show()
        }

        return binding.root

//        val intent = intent
//        val shoppingCart = intent.getSerializableExtra(MainActivity.EXTRA_SHOPPING_CART) as ArrayList<String>
//
//        if (shoppingCart.size > 0) {
//            binding.noItemsText.visibility = View.GONE
//        } else {
//            binding.noItemsText.visibility = View.VISIBLE
//        }
//
    }
}